examples/tests/microbit_v2/main.c

```diff
@@ -0,0 +1,163 @@
+// Micro:bit v2 testing application.
+// This app is modeled after the Hail test app by bradjc.
+
+#include <stdio.h>
+
+#include <adc.h>
+#include <alarm.h>
+#include <ble.h>
+#include <button.h>
+#include <buzzer.h>
+#include <gap.h>
+#include <gpio.h>
+#include <led.h>
+#include <ninedof.h>
+#include <sound_pressure.h>
+#include <temperature.h>
+#include <timer.h>
+#include <tock.h>
+
+
+static int check_err(int retval, const char* func_name) {
+  if (retval < RETURNCODE_SUCCESS) {
+    printf("Function: %s returned error: %s\n", func_name, tock_strrcode(retval));
+    tock_exit(1);
+  }
+  return retval;
+}
+
+
+static bool tone_complete = true;
+static void tone_callback(void) {
+  // allow additional notes to be played
+  tone_complete = true;
+}
+
+static void light_leds(int leds[], int len) {
+  for (int i = 0; i < len; i++) {
+    check_err(led_on(leds[i]), "led_on");
+  }
+}
+
+static void button_callback(int                            btn_num,
+                            int                            button_pressed,
+                            __attribute__ ((unused)) int   arg2,
+                            __attribute__ ((unused)) void *ud) {
+
+  // Do not run if this was a button release or if tone is not finished
+  if (!button_pressed || !tone_complete) {
+    return;
+  }
+
+  // Play a note and make an LED pattern appear
+  tone_complete = false;
+  if (btn_num == 0) {
+    check_err(tone(NOTE_C4 * 3, 500, tone_callback), "tone");
+    int leds[10] = {2, 6, 8, 11, 12, 13, 16, 18, 21, 23};
+    light_leds(leds, 10);
+  } else if (btn_num == 1) {
+    check_err(tone(NOTE_E4 * 3, 500, tone_callback), "tone");
+    int leds[11] = {1, 2, 6, 8, 11, 12, 13, 16, 18, 21, 22};
+    light_leds(leds, 11);
+  } else if (btn_num == 2) {

```

Comment:
Is button 3 the logo? This might not work as expected on every board as the touch logo is not really a button.

Reply:
Until just now, I had assumed that was implemented through some kind of capacitive touch indirection layer. But it doesn't seem like it is: https://github.com/tock/tock/blob/master/boards/microbit_v2/src/main.rs#L261

Amusingly it did work on the board I tested this with. Although I agree it probably shouldn't be relied on.